apps/system/js/audio_channel_manager.js

```diff
@@ -0,0 +1,370 @@
+   * It will allow Music app to play, and pause FM app.
+   *
+   * @class AudioChannelManager
+   */
+  var AudioChannelManager = function() {};
+
+  AudioChannelManager.EVENTS = [
+    'audiochannelstatechanged',
+    'hierarchytopmostwindowchanged',
+    'windowcreated',
+    'windowterminated'
+  ];
+
+  AudioChannelManager.SETTINGS = [
+    'vibration.enabled'
+  ];
+
+  AudioChannelManager.SUB_MODULES = [
+    'AudioChannelPolicy'
+  ];
+
+  BaseModule.create(AudioChannelManager, {
+    name: 'AudioChannelManager',
+    DEBUG: false,
+    // A map contains playing audio channels.
+    _activeAudioChannels: null,
+    // A map contains interrupted audio channels.
+    _interruptedAudioChannels: null,
+    // An array contains audio channels could be resumed
+    // when any other audio channel ends.
+    _audioChannelsResumeWhenOtherEnds: null,
+    // The most top app window.
+    _topMostWindow: null,
+    // The value of the vibration.enabled settings.
+    _isVibrateEnabled: true,
+
+    /**
+     * Initial the module.
+     */
+    _start: function() {
+      this._activeAudioChannels = new Map();
+      this._interruptedAudioChannels = new Map();
+      this._audioChannelsResumeWhenOtherEnds = [];
+      this.readSetting('vibration.enabled').then((value) => {
+        this._isVibrateEnabled = value;
+      });
+      this.debug('Start Audio Channel Manager');
+    },
+
+    /**
+     * Handle the audio chanel when it is active or in inactive.
+     *
+     * @param {Event} evt The event to handle.
+     */
+    _handle_audiochannelstatechanged: function(evt) {
+      var audioChannel = evt.detail.audioChannel;
+      this.debug('Audio channel state is ' + audioChannel.states.active);
+      this._manageAudioChannels(audioChannel);
+    },
+
+    /**
+     * Handle the audio chanel when the app is in foreground or background.
+     */
+    _handle_hierarchytopmostwindowchanged: function() {
+      if (this._topMostWindow) {
+        // Normal channel could not play in background.
+        this.debug(this._topMostWindow.name + ' is closed');
+        var audioChannel = this._topMostWindow.getAudioChannel('normal');
+        if (audioChannel && audioChannel.states.playing) {
+          this._handleAudioChannel({
+            isAllowedToPlay: false,
+            audioChannel: audioChannel
+          })
+          .then(this._resumeBackgroundAudioChannel.bind(this));
+        }
+      }
+      this._topMostWindow = Service.query('getTopMostWindow');
+      // Resume the foreground app's audio channels.
+      this.debug(this._topMostWindow.name + ' is opened');
+      this._resumeForegroundAudioChannels(this._topMostWindow);
+    },
+
+    /**
+     * Register the window's audio channels to be managed
+     * by AudioChannelManager when the window is created.
+     *
+     * @param {Event} evt The event to handle.
+     */
+    _handle_windowcreated: function(evt) {
+      var app = evt.detail;
+      app.registerAudioChannels();
+    },
```

Comment:
Sure.
See it in https://github.com/mozilla-b2g/gaia/pull/26512/files#r27850533.